server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -262,10 +264,14 @@ public ParsedDocument createDeleteTombstoneDoc(String index, String type, String
         return documentParser.parseDocument(emptySource, deleteTombstoneMetadataFieldMappers).toTombstone();
     }
 
-    public ParsedDocument createNoopTombstoneDoc(String index) throws MapperParsingException {
+    public ParsedDocument createNoopTombstoneDoc(String index, String reason) throws MapperParsingException {
         final String id = ""; // _id won't be used.
-        final SourceToParse emptySource = SourceToParse.source(index, type, id, new BytesArray("{}"), XContentType.JSON);
-        return documentParser.parseDocument(emptySource, noopTombstoneMetadataFieldMappers).toTombstone();
+        final SourceToParse sourceToParse = SourceToParse.source(index, type, id, new BytesArray("{}"), XContentType.JSON);
+        final ParsedDocument parsedDoc = documentParser.parseDocument(sourceToParse, noopTombstoneMetadataFieldMappers).toTombstone();
+        // Store the reason of a noop as a raw string in the _source field
+        final BytesRef byteRef = new BytesRef(reason);
```

Comment:
I'm doubting about this - should we always make it valid JSON : `{ "_reason": "bla" }` ? I think it might surprising to find non json in the source field.